source/Nevermore/Advanced/QueryBuilder.cs

```diff
@@ -552,11 +552,9 @@ async Task<(List<TRecord>, int)> ToListWithCountAsyncCte(int skip, int take, Can
             int total = 0;
             var stream = readQueryExecutor.StreamAsync(selectSource.GenerateSql(), parmeterValues, map =>
             {
-                if (total == 0)
-                {
-                    total = map.Read(reader => int.Parse(reader[countColumnName].ToString()));
-                }
-                return map.Map<TRecord>(string.Empty);
+                var result = map.Map<TRecord>(string.Empty);
+                total = map.Read(reader => int.Parse(reader[countColumnName].ToString()));
+                return result;
```

Comment:
This is necessary because the `SequentialAccess` flag requires columns to be read in order. Reading the count column first breaks this rule.

This is a breaking change, and I'm inclined to remove it and address it at a later date.